src/player.js

```diff
@@ -399,6 +407,47 @@ export default class Player extends FakeEventTarget {
    * @private
    */
   _aspectRatio: ?string;
+  /**
+   * An intersection observer instance to track the player visibility.
+   * @type {IntersectionObserver}
+   * @private
+   */
+  _intersectionObserver: window.IntersectionObserver;
+
+  /**
+   * Whether the player is visible in the page scroll view
+   * @type {boolean}
+   * @private
+   */
+  _isVisibleInScroll: boolean;
+
+  /**
+   * Whether the player browser tab is active or not
+   * @type {boolean}
+   * @private
+   */
+  _isTabVisible: boolean;
+
+  /**
+   * Whether the player browser tab is active and in the scroll view
+   * @type {boolean}
+   * @private
+   */
+  _isVisible: boolean;
+
+  /**
+   * Whether the player is in floating mode
+   * @type {boolean}
+   * @private
+   */
+  _isFloating: boolean = false;
```

Comment:
if core player is aware of floating state which is from a plugin we are breaking isolation of features

Reply:
Okay... But this is part of the product request...
How else can I implement this?
Why is this breaking isolation? I am not relying of the plugin to be part. Either it is and then I consider it or not.

Reply:
it is like saying not to check PIP because it is implemented by the browser.

Reply:
Core doesn't know about floating so now it needs to.
If you want to have some generic way to signal core player to ignore condition then this is a generic api that can then be used for other plugins if they want to tell the feature to not trigger

Reply:
And per pip, it is not the same as we implement pip internally while floating is not

Reply:
Not relevant anymore. Code was refactored